feat(glance): Add Ctrl/Cmd+O shortcut to expand Glance; user-configurable via Keyboard Shortcuts; tooltips

KyleKincer opened the pull request:
Adds a default, user-configurable shortcut to expand Glance, and improves Glance button tooltips.

Changes
- Default shortcut: Ctrl+O (Cmd+O on macOS) to expand Glance to a full tab.
- User-configurable: Shortcut is exposed in Settings → Keyboard Shortcuts (under Other).
- Context-aware: The key is enabled only while a Glance is active to avoid hijacking the Open File action when no Glance is open.
- Safety: `fullyOpenGlance()` no-ops when no Glance is active.
- Tooltips: Glance overlay buttons now display helpful tooltips with the current shortcut.
  - Close shows “Esc” even when there isn’t a <key> binding (since Escape is handled in the Glance actor).
  - Expand/Split reflect the user-configured keybinding dynamically.

Implementation
- Keyboard shortcuts: bump KBS version and add migration to include a default `zen-glance-expand` key bound to `cmd_zenGlanceExpand`.
- Dynamic enable/disable: toggle the expand key element while opening/closing Glance.
- Tooltip builder: derive shortcut text from `<key>` elements; fall back to “Esc” for Close.

Related discussion
- https://github.com/zen-browser/desktop/discussions/5079

Testing
- Manual: open Glance, press Ctrl/Cmd+O to expand. With no Glance open, Ctrl/Cmd+O continues to open the file dialog. Tooltips show the expected shortcuts.
- Automated: `python3 scripts/run_tests.py glance/browser_glance_expand.js` (and other glance tests).

KyleKincer committed:
feat(glance): Add Ctrl/Cmd+O shortcut to expand Glance; user-configurable via Keyboard Shortcuts; disable when inactive

- Adds default shortcut / mapped to  via Zen keyboard shortcuts system
- Enables key only while a Glance is active to avoid hijacking Open File
- Bumps KBS version and migrates defaults; adds guard in fullyOpenGlance()
- No l10n string added (falls back to command label in settings)

mr-cheffy commented on src/zen/glance/ZenGlanceManager.mjs:
```diff
@@ -116,6 +119,52 @@
+          el.setAttribute('aria-label', text);
+        };
+
+        setTooltip(closeBtn, 'Close Glance', 'cmd_zenGlanceClose');
```
These should be translated

mr-cheffy commented on src/zen/glance/ZenGlanceManager.mjs:
```diff
@@ -116,6 +119,52 @@
+        const expandBtn = newButtons.querySelector('.zen-glance-sidebar-open');
+        const splitBtn = newButtons.querySelector('.zen-glance-sidebar-split');
+
+        const buildShortcutText = (keyElem) => {
```
This should rather be moved into a shortcut generic space, like on nsZenKeyboardShortcut...

KyleKincer commented:
@mr-cheffy thanks for the feedback, updated :+1:

mr-cheffy commented on src/zen/glance/ZenGlanceManager.mjs:
```diff
@@ -26,6 +26,9 @@
       Services.obs.addObserver(this, 'quit-application-requested');
+
+      // Disable expand shortcut by default when no glance is active
+      document.getElementById('zen-glance-expand')?.setAttribute('disabled', true);
```
Why is it being set as disabled if we already have the guard check at the start of "fulylExpandGlance"?

mr-cheffy reviewed (changes requested):
I'd honestly just merge the shortcut part and make a different PR for tooltips and text...

KyleKincer commented:
@mr-cheffy thanks again-- made those changes and reverted the tooltips and internationalizations-- Will submit that separately. 
Weirdly, it was working with these changes initially, but today in my testing the shortcut was being consumed by Firefox's default Open File command. I added some code to consume the shortcut before Firefox can and it's working again-- let me know if you prefer a different approach

mr-cheffy reviewed (changes requested):
Instead of this, we should change the `accel+O` to another keybind as this is not really used...

KyleKincer commented:
@mr-cheffy Change the Expand Glance shortcut, or the Open File shortcut from `accel+O`?

mr-cheffy commented:
Open file shortcut. Maybe add an extra combination like ctrl+shift+o? (I can't check if that's taken as well right now)

KyleKincer committed:
fix: remove hack to steal default shortcut

KyleKincer committed:
fix: modify Open File default shortcut during migration

KyleKincer committed:
cleanup: remove unnecessary changes to ZenGlanceManager

KyleKincer commented:
Sounds good, done. I also removed the changes to `ZenGlanceManager` as I don't think they were necessary at all

mr-cheffy reviewed (approved):
Looks good! if you try to do that shortcut without being on glance, does any error occour?

nexxai commented:
> Instead of this, we should change the `accel+O` to another keybind as this is not really used...

I can't speak for anyone else, but I've been looking for this keyboard shortcut since coming over from Arc, and CMD+O is what they used there, so if it can default to that (or at least have it as a configurable option), that would be great.

KyleKincer commented:
No user-facing errors for sure. I will confirm nothing in the console when I'm home from my day job 👍

mr-cheffy commented:
What about experimenting with simply removing the shortcut to open files?

KyleKincer committed:
remove open file shortcut

KyleKincer committed:
fix: error when no glance open

KyleKincer commented:
@mr-cheffy Good call, I doubt people would miss it. We may need to vibe check after release. Current PR has it completely removed. There was an error triggering the shortcut with no glance active-- Added the minimal fix for that as well

mr-cheffy commented:
Good then! I'll merge to twilight this night and we'll see if there are any regressions. Thanks!
